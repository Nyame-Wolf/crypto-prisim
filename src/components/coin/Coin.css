.container {
  background-color: #021e00;
  width: 100%;
  aspect-ratio: 2 / 1;
  display: flex;
}

.coin {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 40px;
}
