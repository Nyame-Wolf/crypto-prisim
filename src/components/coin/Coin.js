import PropTypes from 'prop-types';
import './Coin.css';

const Coin = (props) => {
  const { coin } = props;

  return (
    <div className="coin">
      <p className="coin-name">{coin.name}</p>
      <small>{`$${coin.price_usd}`}</small>
    </div>
  );
};

export default Coin;

Coin.propTypes = {
  coin: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    symbol: PropTypes.string,
    rank: PropTypes.number,
    price_usd: PropTypes.string,
  }).isRequired,
};
