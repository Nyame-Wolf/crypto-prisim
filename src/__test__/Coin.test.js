import React from 'react';
import { render } from '@testing-library/react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Coin from '../components/coin/Coin';

describe('Coin test', () => {
  test('snapshot for coin', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Coin
            coin={{
              id: '90',
              symbol: 'BTC',
              name: 'Bitcoin',
              rank: 1,
              price_usd: '20142.07',
            }}
          />
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});

Coin.propTypes = {
  coin: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    symbol: PropTypes.string,
    rank: PropTypes.number,
    price_usd: PropTypes.string,
  }).isRequired,
};
