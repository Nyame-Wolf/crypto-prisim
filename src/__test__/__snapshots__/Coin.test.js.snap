// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin test snapshot for coin 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="coin"
      >
        <p
          class="coin-name"
        >
          Bitcoin
        </p>
        <small>
          $20142.07
        </small>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="coin"
    >
      <p
        class="coin-name"
      >
        Bitcoin
      </p>
      <small>
        $20142.07
      </small>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
